test/stream.js

```diff
@@ -693,3 +699,70 @@ test('no idle after close', async function (t) {
     })
     .connect(socket, 2, socket.address().port)
 })
+
+test('localHost, localFamily and localPort', async function (t) {
+  t.plan(6)
+
+  const udx = new UDX()
+
+  const socket = udx.createSocket()
+  socket.bind(0)
+
+  const stream = udx.createStream(1)
+
+  t.is(stream.localHost, null)
+  t.is(stream.localFamily, 0)
+  t.is(stream.localPort, 0)
+
+  stream.on('connect', function () {
+    t.is(stream.localHost, '0.0.0.0')
+    t.is(stream.localFamily, 4)
+    t.is(typeof stream.localPort, 'number')
+
+    stream.destroy()
+    socket.close()
+  })
+
+  stream.connect(socket, 2, socket.address().port)
```

Comment:
You need to pass the remote host here also “127.0.0.1”, otherwise this is undefined behaviour

Reply:
Would not be aproppiate to also be able to use `socket.address().host` as an argument there? As `address().port` was used
But `address().host` returns `0.0.0.0`

Reply:
Ya, you have to hardcode 127.0.0.1 here

Reply:
What is the difference with `emit connect` test?

Reply:
that doesn't pass 127.0.0.1? that's a bug then :) can you link?

Reply:
we should fix

Reply:
But wait, the host is defaulted to `127.0.0.1`:
https://github.com/hyperswarm/libudx/blob/a4d4befbfa859f1dc375f00c4e44091f5fa79404/lib/stream.js#L107
So no need to pass that address in those cases?

Reply:
ah sorry didnt know. then fine to omit it